### ISSUE TITLE
E0602 false positive

### ISSUE THREAD

**Issue by seehuhn** [PyCQA/pylint#2989]
Pylint wrongly reports an undefined variable when run on the following code:

```
"""Demonstrate a pylint bug."""

import numpy as np

class Funny:

    """Class docstring."""

    @staticmethod
    def set(dst, dst_mask, src, src_mask, batch):
        """Method docstring."""
        for i, (_, a, b, c) in enumerate(batch):
            if np.any(dst_mark[b:c]):
                raise NotImplementedError()
            dst[b:c] = src[b-a:c-a, i]
            dst_mask[b:c] = src_mask[b-a:c-a, i]
```

The variable `dst_mark` is defined, since it is the second argument of the method, but pylint reports is as being undefined:

```
>>> pylint err.py
************* Module err
err.py:13:22: E0602: Undefined variable 'dst_mark' (undefined-variable)

------------------------------------------------------------------
Your code has been rated at 3.75/10 (previous run: 3.75/10, +0.00)

>>> pylint --version
pylint 2.3.1
astroid 2.2.5
Python 3.7.3 (default, Mar 27 2019, 09:23:39)
[Clang 10.0.0 (clang-1000.11.45.5)]
>>> uname -a
Darwin *** 17.7.0 Darwin Kernel Version 17.7.0: Wed Apr 24 21:17:24 PDT 2019; root:xnu-4570.71.45~1/RELEASE_X86_64 x86_64

```

**Comment by seehuhn** [comment]
Sorry about that!  Clearly the problem here was just that I can't spell, and `pylint` is fine.  My apologies ...

**Comment by PCManticore** [comment]
No worries @seehuhn !
